### ISSUE TITLE
Cannot set working plane to a given Arch_Floor

### ISSUE THREAD

**Issue by furgo16** [FreeCAD/FreeCAD#11157]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Since a few days, I'm no longer able to set the working plane to a given Arch_Floor. Before, with the Arch workbench active, I used to be able to select a level (Arch_Floor) defined in the TreeView, then click on the working plane selector to set the current working plane to that level.

Now, on the step of clicking on the working plane selector, I quite often get the following error on the Report panel, with the subsequent inability to set the working plane:

```
21:22:28  Running the Python command 'Draft_SelectPlane' failed:
Traceback (most recent call last):
  File "/snap/freecad/811/usr/Mod/Draft/draftguitools/gui_selectplane.py", line 73, in Activated
    App.activeDraftCommand.finish()
  File "/snap/freecad/811/usr/Mod/Draft/draftguitools/gui_selectplane.py", line 152, in finish
    if self.call:
```

Unfortunately, it is not always reproducible.

### Full version info

```shell
[code]
OS: Ubuntu Core 22 (ubuntu:GNOME/ubuntu-xorg)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.34735 (Git) Snap 811
Build type: Release
Branch: main
Hash: 8636dd1848cbfaa023ecafe8aeb5e43bc50fe50f
Python 3.10.12, Qt 5.15.10, Coin 4.0.0, Vtk 7.1.1, OCC 7.7.1
Installed mods: 
  * BIM 2021.12.0
[/code]
```

### Subproject(s) affected?

Draft

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Roy-043** [comment]
There are 2 issues here:
1. The Placement of a BuildingPart is ignored. This has already been fixed with #11144.
2. The Draft_SelectPlane command is not correctly finished if there is a pre-selection. I'll submit a PR for that.

**Comment by furgo16** [comment]
Thanks so much for the quick fix and your work in FreeCAD!
